### PR TITLE
refactor(types): deduplicate JSDoc

### PR DESCRIPTION
**Pull request by jsejcksn**
Ref: https://github.com/iambumblehead/esmock/pull/163#pullrequestreview-1120344255

### CODE REVIEW

File: src/esmock.d.ts [hunk 1]
```diff
@@ -1,71 +1,85 @@
-type mocks = Record<string, any>
+type MockMap = { [specifier: string]: any };
 
-type opts = {
-  strict?: boolean | undefined,
-  purge?: boolean | undefined,
-  isModuleNotFoundError?: boolean | undefined
-}
+type Options = {
```
Comment (jsejcksn) [inline src/esmock.d.ts:3]: The consumer can alias a shorter name if desired. Clear semantics are useful.

**Comment by jsejcksn** [comment]
@iambumblehead I left a few observations about the changes — mostly just artifacts for you and others who happen to review this PR in the future.

File: src/esmock.d.ts [hunk 2]
```diff
@@ -1,71 +1,85 @@
-type mocks = Record<string, any>
+type MockMap = { [specifier: string]: any };
```
[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/esmock.d.ts:1]: Thanks for this reasoned perspective and change. The string version is more readable for me.

File: src/esmock.d.ts (hunk 1 above)
Comment (iambumblehead) [reply src/esmock.d.ts:3]: okay thank you

File: src/esmock.d.ts [hunk 3]
```diff
@@ -1,71 +1,85 @@
...
+    gdefs?: MockMap,
+    opts?: Options,
+  ): any;
+  (
+    modulePath: string,
+    defs?: MockMap,
+    gdefs?: MockMap,
+    opts?: Options,
+  ): any;
+};
```
[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/esmock.d.ts:42]: This looks really great and I'll probably use this in the future. When searching the internet over the weekend, I did not find this anywhere and maybe there are not many who know about this. Thanks

**Comment by iambumblehead** [comment]
I'll play with the `@see` directive per your advice and maybe make a follow up PR sometime this week

**Pull request merged**